go-controller/pkg/ovn/gateway_init.go

```diff
@@ -329,25 +329,31 @@ func (oc *DefaultNetworkController) gatewayInit(nodeName string, clusterIPSubnet
 			IPPrefix: hostSubnet.String(),
 			Nexthop:  gwLRPIP[0].String(),
 		}
-		p := func(item *nbdb.LogicalRouterStaticRoute) bool {
-			return item.Nexthop == lrsr.Nexthop && item.IPPrefix == lrsr.IPPrefix && item.Policy != nil && *item.Policy == *lrsr.Policy
-		}
+
 		if config.Gateway.Mode != config.GatewayModeLocal {
+			p := func(item *nbdb.LogicalRouterStaticRoute) bool {
+				return item.IPPrefix == lrsr.IPPrefix && item.Policy != nil && *item.Policy == *lrsr.Policy
+			}
 			// If migrating from local to shared gateway, let's remove the static routes towards
 			// management port interface for the hostSubnet prefix before adding the routes
 			// towards join switch.
 			mgmtIfAddr := util.GetNodeManagementIfAddr(hostSubnet)
 			oc.staticRouteCleanup([]net.IP{mgmtIfAddr.IP})
 
-			err := libovsdbops.CreateOrUpdateLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter,
+			err := libovsdbops.CreateOrUpdateUniqueLogicalRouterStaticRoutesWithPredicate(oc.nbClient, types.OVNClusterRouter,
 				&lrsr, p)
 			if err != nil {
 				return fmt.Errorf("error creating static route %+v in GR %s: %v", lrsr, types.OVNClusterRouter, err)
 			}
 		} else if config.Gateway.Mode == config.GatewayModeLocal {
 			// If migrating from shared to local gateway, let's remove the static routes towards
-			// join switch for the hostSubnet prefix before adding the routes
-			// towards management port which is done in syncNodeManagementPort.
+			// join switch for the hostSubnet prefix
+			// Note syncManagementPort happens before gateway sync so only remove things pointing to join subnet
```

Comment:
note this probably isn't necessary since the sync management code will make sure there is only 1 route per prefix with src-ip, but it doesn't hurt to leave this functionality as extra cleanup